api/desecapi/tests/test_rrsets.py

```diff
@@ -239,14 +248,155 @@ def test_create_my_rr_sets_empty_payload(self):
         response = self.client.post_rr_set(self.my_empty_domain.name)
         self.assertContains(response, 'No data provided', status_code=status.HTTP_400_BAD_REQUEST)
 
-    def test_create_my_rr_sets_unknown_type(self):
-        for _type in ['AA', 'ASDF']:
-            with self.assertPdnsRequests(
-                    self.request_pdns_zone_update_unknown_type(name=self.my_domain.name, unknown_types=_type)
-            ):
-                response = self.client.post_rr_set(self.my_domain.name, records=['1234'], ttl=3660, type=_type)
+    def test_create_my_rr_sets_canonical_content(self):
+        # TODO fill in more examples
+        datas = [
+            # record type: (non-canonical input, canonical output expectation)
+            ('A', ('127.0.000.1', '127.0.0.1')),
+            ('AAAA', ('0000::0000:0001', '::1')),
+            ('AFSDB', ('02 turquoise.femto.edu.', '2 turquoise.femto.edu.')),
+            ('CAA', ('', '')),
+            ('CDNSKEY', ('', '')),
+            ('CDS', ('', '')),
+            ('CERT', ('', '')),
+            ('CNAME', ('', '')),
+            ('DHCID', ('', '')),
+            ('DLV', ('', '')),
+            ('DS', ('', '')),
+            ('EUI48', ('', '')),
+            ('EUI64', ('', '')),
+            ('HINFO', ('', '')),
+            ('IPSECKEY', ('', '')),
+            ('KEY', ('', '')),
+            ('KX', ('', '')),
+            ('LOC', ('', '')),
+            ('MAILA', ('', '')),
+            ('MAILB', ('', '')),
+            ('MINFO', ('', '')),
+            ('MR', ('', '')),
+            ('MX', ('', '')),
+            ('NAPTR', ('', '')),
+            ('NS', ('', '')),
+            ('NSEC', ('', '')),
+            ('NSEC3', ('', '')),
+            ('OPENPGPKEY', ('', '')),
+            ('PTR', ('', '')),
+            ('RP', ('', '')),
+            ('SPF', ('', '')),
+            ('SRV', ('', '')),
+            ('SSHFP', ('', '')),
+            ('TKEY', ('', '')),
+            ('TLSA', ('3 0001 1 000AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA', '3 1 1 000aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa')),
+            ('TSIG', ('', '')),
+            ('TXT', ('"foobar"', 'foobar')),
+            ('URI', ('', '')),
+            ('WKS', ('', '')),
+        ]
+        for t, (record, canonical_record) in datas:
+            if not record:
+                continue
+            data = {'records': [record], 'ttl': 3660, 'type': t, 'subname': ''}
+            with self.assertPdnsRequests(self.requests_desec_rr_sets_update(name=self.my_empty_domain.name)):
+                response = self.client.post_rr_set(self.my_empty_domain.name, **data)
+                self.assertContains(response, canonical_record, status_code=status.HTTP_201_CREATED)
+            with self.assertPdnsRequests(self.requests_desec_rr_sets_update(name=self.my_empty_domain.name)):
+                response = self.client.delete_rr_set(self.my_empty_domain.name, subname='', type_=t)
```

Comment:
`type_` could be made `type` (without shadowing anything) if the function definition is changed from

```
    def delete_rr_set(self, domain_name, subname, type_):
        return self.delete(
            self.reverse('v1:rrset@', name=domain_name, subname=subname, type=type_)
        )
```

to

```
    def delete_rr_set(self, domain_name, *args, **kwargs):
        return self.delete(
            self.reverse('v1:rrset@', name=domain_name, *args, **kwargs)
        )
```

But I recognize this has nothing to do with this PR.

Reply:
There's a whole bunch of functions with similar interface, changing all of them isn't worth the small benefit, just changing this one would make the code style worse. So, I'll opt for being lazy